SlevomatCodingStandard/Sniffs/ControlStructures/AbstractControlStructureSpacingSniff.php

```diff
@@ -0,0 +1,388 @@
+<?php declare(strict_types = 1);
+
+namespace SlevomatCodingStandard\Sniffs\ControlStructures;
+
+use Exception;
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Util\Tokens;
+use SlevomatCodingStandard\Helpers\SniffSettingsHelper;
+use SlevomatCodingStandard\Helpers\TokenHelper;
+use SlevomatCodingStandard\Sniffs\Namespaces\UndefinedKeywordTokenException;
+use function abs;
+use function array_key_exists;
+use function array_map;
+use function constant;
+use function count;
+use function defined;
+use function in_array;
+use function sprintf;
+use function strlen;
+use function substr;
+use function substr_count;
+use const T_ANON_CLASS;
+use const T_CASE;
+use const T_CATCH;
+use const T_CLOSE_CURLY_BRACKET;
+use const T_CLOSURE;
+use const T_COLON;
+use const T_COMMENT;
+use const T_DEFAULT;
+use const T_DO;
+use const T_ELSE;
+use const T_ELSEIF;
+use const T_FINALLY;
+use const T_FOR;
+use const T_FOREACH;
+use const T_IF;
+use const T_OPEN_CURLY_BRACKET;
+use const T_OPEN_SHORT_ARRAY;
+use const T_OPEN_TAG;
+use const T_SEMICOLON;
+use const T_SWITCH;
+use const T_TRY;
+use const T_WHILE;
+use const T_WHITESPACE;
+use const T_YIELD;
+use const T_YIELD_FROM;
+
+abstract class AbstractControlStructureSpacingSniff implements Sniff
+{
+
+	public const CODE_INCORRECT_LINES_COUNT_BEFORE_CONTROL_STRUCTURE = 'IncorrectLinesCountBeforeControlStructure';
+	public const CODE_INCORRECT_LINES_COUNT_BEFORE_FIRST_CONTROL_STRUCTURE = 'IncorrectLinesCountBeforeFirstControlStructure';
+	public const CODE_INCORRECT_LINES_COUNT_AFTER_CONTROL_STRUCTURE = 'IncorrectLinesCountAfterControlStructure';
+	public const CODE_INCORRECT_LINES_COUNT_AFTER_LAST_CONTROL_STRUCTURE = 'IncorrectLinesCountAfterLastControlStructure';
+
+	/** @var int */
+	public $linesCountBeforeControlStructure = 1;
+
+	/** @var int */
+	public $linesCountBeforeFirstControlStructure = 0;
+
+	/** @var int */
+	public $linesCountAfterControlStructure = 1;
+
+	/** @var int */
+	public $linesCountAfterLastControlStructure = 0;
+
+	/** @var bool */
+	public $allowSingleLineYieldStacking = true;
```

Comment:
Move this option to `JumpStatementsSpacing` please. I'm ok if you move only the option :)

Reply:
Now that I think about it it would be better to move it all ... I'm also not sure how would I separate the option and the code :D